pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -1156,6 +1156,7 @@ func initPublications(zedagentCtx *zedagentContext) {
 		log.Fatal(err)
 	}
 	getconfigCtx.pubBaseOsConfig.ClearRestarted()
+	populateBaseOsConfig(getconfigCtx)
```

Comment:
In general I do not see any problems in inhaling config earlier (here I can be mistaken). But calling `populateBaseOsConfig()` from the `initPublications()` is misleading: you do not expect that a config will be read and something published on that path. 

As a suggestion: why not to extend the `maybeLoadBootstrapConfig()` (which happens a few lines after the `initPublications` is called) and inhale either whole bootstrap or whole last config ?

Reply:
Noted :+1: That makes sense, but I will have to redo the PR given the reason in @eriknordmark's comment